app/controlplane/internal/usercontext/attjwtmiddleware/attmiddleware.go

```diff
@@ -191,6 +230,29 @@ func WithJWTMulti(opts ...JWTOption) middleware.Middleware {
 							continue
 						}
 
+						// If federated verification is enabled, we try to get the information remotely
+						if o.federatedVerificationURL != "" {
```

Comment:
I might be missing where the decision is made, but Is this logic called always, no matter if the token belongs to the federated provider?

Reply:
That's correct, the federated provider is the one that will figure out if it understands the provided token

Reply:
Cannot this be a performance issue?  Doesn't it have an `issuer` claim that can be used to early detect if the token is federated or not?

Reply:
Calling an external URL in every single request to check a token that might be already been processed and understood. Unless I'm missing something, of course.